File: gfg/Can Place Flowers.cpp
//https://leetcode.com/problems/can-place-flowers/

class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int p = flowerbed.size();
        int c=0;
        
        for(int i = 0 ; i < flowerbed.size();i++){
            if(flowerbed[i]==0) {
                int pre =(i== 0 || flowerbed[i - 1] == 0) ? 0 : 1;
                int ne = (i== flowerbed.size()-1 || flowerbed[i + 1] == 0) ? 0 : 1;
                
                
                if(pre==0 && ne==0){
                    flowerbed[i]=1;
                    c++;
                }
            }
        }
        return  c>=n;
        // return false ;
    }
};

File: LeetCode/Can Place Flowers.rb
# https://leetcode.com/problems/can-place-flowers/description/

# @param {Integer[]} flowerbed
# @param {Integer} n
# @return {Boolean}
def can_place_flowers(flowerbed, n)
    count = 0 
    (0..flowerbed.length-1).each do |i|
        if flowerbed[i] == 0 
            previous = flowerbed[i-1] == 0 || i==0 ? true : false
            nex = flowerbed[i+1] == 0 || i == flowerbed.length-1  ? true : false
            if previous == true and nex == true
                count += 1
                flowerbed[i] = 1
            end
        end
    end
    return count >= n
end